src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            fileName = fileName + ".tar.gz";
-            filePath = filePath + ".tar.gz";
-
-            int fileSize = Utils::getFileSize(filePath);
-            std::string fileLength = to_string(fileSize);
-            // send file name
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            line = (data);
-            if (line.compare(JasmineGraphInstanceProtocol::SEND_FILE_NAME) == 0) {
-                write(connFd, fileName.c_str(), fileName.size());
-                instance_logger.log("Sent : File name " + fileName, "info");
+                string egonetString;
 
-                bzero(data, INSTANCE_DATA_LENGTH + 1);
-                read(connFd, data, INSTANCE_DATA_LENGTH);
-                line = (data);
-                // send file length
-                if (line.compare(JasmineGraphInstanceProtocol::SEND_FILE_LEN) == 0) {
-                    instance_logger.log("Received : " + JasmineGraphInstanceProtocol::SEND_FILE_LEN, "info");
-                    write(connFd, fileLength.c_str(), fileLength.size());
-                    instance_logger.log("Sent : File length in bytes " + fileLength, "info");
+                int partitionID = stoi(workerSocketPair[2]);
+                result_wr = write(sockfd, std::to_string(partitionID).c_str(), std::to_string(partitionID).size());
 
-                    bzero(data, INSTANCE_DATA_LENGTH + 1);
-                    read(connFd, data, INSTANCE_DATA_LENGTH);
-                    line = (data);
-                    // send content
-                    if (line.compare(JasmineGraphInstanceProtocol::SEND_FILE_CONT) == 0) {
-                        instance_logger.log("Received : " + JasmineGraphInstanceProtocol::SEND_FILE_CONT, "info");
-                        instance_logger.log("Going to send file through service", "info");
-                        JasmineGraphInstance::sendFileThroughService(serverHostName, stoi(serverHostDataPort), fileName,
-                                                                     filePath);
-                    }
+                if (result_wr < 0) {
+                    instance_logger.log("Error writing to socket", "error");
                 }
-            }
-            int count = 0;
-            while (true) {
-                write(connFd, JasmineGraphInstanceProtocol::FILE_RECV_CHK.c_str(),
-                      JasmineGraphInstanceProtocol::FILE_RECV_CHK.size());
-                instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::FILE_RECV_CHK, "info");
-                instance_logger.log("Checking if file is received", "info");
-                bzero(data, INSTANCE_DATA_LENGTH + 1);
-                read(connFd, data, INSTANCE_DATA_LENGTH);
-                line = (data);
+                instance_logger.log("Sent : Partition ID " + std::to_string(partitionID), "info");
+                bzero(data, 301);
+                read(sockfd, data, 300);
```

Comment:
```suggestion
                read(sockfd, data, INSTANCE_DATA_LENGTH);
```

Reply:
All such occurrences are replaced in next PR

Reply:
Ok